tests/subsys/fs/littlefs/boards/mimxrt1060_evk.overlay

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&is25wp064 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(64)>;
+		};
+		partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 DT_SIZE_M(3)>;
+		};
```

Comment:
As far as I can tell, these are exactly the same as in original board DTS, so they probably do not require redefinition.
With the other partitions: they take space of `slot1_partition`, `scratch_partition` and `storage_partition` from original layout (boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts) so it would be cleaner to remove the these first.

Reply:
change to below does not work for me

```
/*
 * Copyright (c) 2021 NXP
 *
 * SPDX-License-Identifier: Apache-2.0
 */

&is25wp064 {
        partitions {
                compatible = "fixed-partitions";
                #address-cells = <1>;
                #size-cells = <1>;

                partition@310000 {
                        label = "large";
                        reg = <0x00310000 DT_SIZE_M(3)>;
                };
                partition@610000 {
                        label = "image-scratch";
                        reg = <0x00610000 DT_SIZE_K(128)>;
                };
                partition@630000 {
                        label = "small";
                        reg = <0x00630000 DT_SIZE_K(64)>;
                };
                partition@640000 {
                        label = "medium";
                        reg = <0x00640000 DT_SIZE_K(960)>;
                };
        };
};
```
I meet below error

```
E: ***** MPU FAULT *****
E:   Instruction Access Violation
E: r0/a1:  0x800034b8  r1/a2:  0x00000000  r2/a3:  0x00000001
E: r3/a4:  0x80000ddd r12/ip:  0x150b0000 r14/lr:  0x6000d6ed
E:  xpsr:  0x21000000
E: Faulting instruction address (r15/pc): 0x80000ddc
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x80003680 (main)
E: Halting system
E: ***** MPU FAULT *****
E:   Instruction Access Violation
E: r0/a1:  0x800034b8  r1/a2:  0x00000000  r2/a3:  0x00000001
E: r3/a4:  0x80000ddd r12/ip:  0x150b0000 r14/lr:  0x6000d6ed
E:  xpsr:  0x21000000
E: Faulting instruction address (r15/pc): 0x80000ddc
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x80003680 (main)
E: Halting system
```

Reply:
This is strange. I do not think that the change in overlay, by removal of duplicate definitions, should affect build in such way.
I will not insist on the overlay change, for the PR, but I think that this requires some investigation.

Reply:
I do a diff on the build folder, and find the mimxrt1060_evk.dts.pre.tmp is not the same, see below.
![Image 2](https://user-images.githubusercontent.com/800198/118266471-3be45280-b4ed-11eb-82ed-5539b577e0f5.png)
and the the zephyr.dts are the same for both build. looks like some intermedia data is used during compilation.

Reply:
This works for me. Perhaps you need to do a mass erase when you program the flash (i.e., `west flash --erase`?

```diff
diff --git a/tests/subsys/fs/littlefs/boards/mimxrt1060_evk.overlay b/tests/subsys/fs/littlefs/boards/mimxrt1060_evk.overlay
new file mode 100644
index 0000000000..94374cbd72
--- /dev/null
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1060_evk.overlay
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&is25wp064 {
+       partitions {
+               compatible = "fixed-partitions";
+               #address-cells = <1>;
+               #size-cells = <1>;
+
+               partition@310000 {
+                       label = "large";
+                       reg = <0x00310000 DT_SIZE_M(3)>;
+               };
+               partition@630000 {
+                       label = "small";
+                       reg = <0x00630000 DT_SIZE_K(64)>;
+               };
+               partition@640000 {
+                       label = "medium";
+                       reg = <0x00640000 DT_SIZE_K(960)>;
+               };
+       };
+};
```

Reply:
if we need --erase for this test, then it will impact our CI testing, shall we keep the former ones?

Reply:
I don't know for sure if we need it yet. Can you try it to see if that allows the test to pass for you?

Reply:
with the concise format, I need add --erase every time, but with the full content format, I just need west flash --runner=pyocd

Reply:
both function works at my side

Reply:
both function works at my side

Reply:
Ok we can keep the full content if that helps testing, but will you please document this in the commit message?

Reply:
ok